java/code/src/com/redhat/rhn/frontend/action/errata/ChannelSetupAction.java

```diff
@@ -65,7 +68,8 @@ public ActionForward execute(ActionMapping mapping,
         clampListBounds(pc, request, user);
 
         //Get the errata object
-        Errata e = requestContext.lookupErratum();
+        Long errataId = requestContext.getParamAsLong(ERRATA_ID);
+        Errata e = ErrataManager.lookupErrata(errataId, requestContext.getCurrentUser());
```

Comment:
might make sense to return an optional here if Errata can be null. Maybe for a next PR

Reply:
Yes, I am adding that to the list for the follow-up cleanup PR.